packages/cicero-core/test/template.js

```diff
@@ -34,6 +36,30 @@ function waitForEvent(emitter, eventType) {
     });
 }
 
+function sign(templateHash, timeStamp, path, password){
```

Comment:
Is this to create a private key? This is hard to understand without some documentation and with the generic `sign` name. How does it relate to the sign function in the template class?

Reply:
No, its not to create a private key. Private key is already present inside the keystore whose path and password is being passed to this function. Since the signatures are unique and depends upon the time they are being signed on. This function signs the hash and timestamp string again and compares that signature with object returned from `signTemplate`.

Reply:
That's really confusing. Why isn't that calling the function in the `template` class then?

Reply:
It is calling both. The `sign` function in the test is used to generated the expected signature object and the `signTemplate` method from the Template class is used to generate the resultant signature object. The test passes if both the signature objects are same.

Reply:
🤔 what's the difference in the two functions?

Reply:
They do the same job. One generates signature in the test.js file. Another does that using the method in the template instance. And then both signatures are compared.

Reply:
hm... then why not call `Template.signTemplate` instead of duplicating the code?

Reply:
the `sign` function has the same logic that `Template.signTemplate` has. So, inside the test, both `sign` and `Template.signTemplate` are called. `sign` function generates the expected signature whereas the `Template.signTemplate` generates the resultant signature. If there is any change in the logic of `Template.signTemplate` it can be captured during the test as it won't match with the signature generated by the auxiliary function `sign` in the test.

Reply:
https://github.com/accordproject/cicero/pull/672#issuecomment-859274932